src/views/tables/OrdersTable/CancelAllOrdersButton.tsx

```diff
@@ -0,0 +1,60 @@
+import { useCallback } from 'react';
+
+import { useDispatch } from 'react-redux';
+import styled, { css } from 'styled-components';
+
+import { ButtonAction, ButtonSize } from '@/constants/buttons';
+
+import { useParameterizedSelector } from '@/hooks/useParameterizedSelector';
+import { useSubaccount } from '@/hooks/useSubaccount';
+
+import { Button } from '@/components/Button';
+
+import { clearAllOrders } from '@/state/account';
+import { calculateHasCancelableOrders } from '@/state/accountCalculators';
+
+type ElementProps = {
+  marketId?: string;
+};
+
+export const CancelAllOrdersButton = ({ marketId }: ElementProps) => {
```

Comment:
nit: maybe consider renaming this component since it's either a `Cancel` or `Clear` button - maybe `RemoveAllOrdersButton`?